Agents/PropertyValueEstimationAgent/propertyvalueestimation/agent/value_estimation.py

```diff
@@ -0,0 +1,212 @@
+                with ts_client.connect() as conn:
+                    ts = ts_client.tsclient.getTimeSeries([prop_price_index_iri], conn)
+                dates = [d.toString() for d in ts.getTimes()]
+                values = [v for v in ts.getValues(prop_price_index_iri)]
+            except Exception as ex:
+                self.logger.error('Error retrieving/unwrapping Property Price Index time series')
+                raise TSException('Error retrieving/unwrapping Property Price Index time series') from ex
+
+            # Create UKHPI series with conditioned date index
+            ukhpi = pd.Series(index=dates, data=values)
+            ukhpi = ukhpi.astype(float)
+            ukhpi.index = pd.to_datetime(ukhpi.index, format=TIME_FORMAT_LONG)
+            ukhpi.sort_index(ascending=False, inplace=True)
+            ukhpi.index = ukhpi.index.strftime(TIME_FORMAT_SHORT)
+
+            # 2) Retrieve previous sales transaction details for previous transaction IRI
+            #    and adjust to current market value
+            res = self.sparql_client.get_transaction_details(transaction_iri)
+            ukhpi_now = ukhpi.iloc[0]
+            ukhpi_old = ukhpi[res['date']]
+            market_value = res['price'] * ukhpi_now / ukhpi_old
+
+        # Prio 2: Otherwise assess market value based on FloorArea and AveragePricePerSqm
+        elif avgsqm_price_iri and floor_area_iri and not market_value:
+            # NOTE: To ensure availability of AvgSqmPrice (i.e. derivation being computed by
+            #       AvgSqmPrice Agent), AvgSqmPrice should be marked up as Synchronous Derivation
+            res = self.sparql_client.get_floor_area_and_avg_price(floor_area_iri)
+            market_value = res['floor_area'] * res['avg_price']
+
+        if market_value:
+            # Round property market value to full kGBP
+            market_value = round(market_value/1000)*1000
+            # Create instantiation/update triples
+            market_value_iri = KB + 'AmountOfMoney_' + str(uuid.uuid4())
+            # Create rdflib graph with update triples 
+            g = self.sparql_client.instantiate_property_value(graph=g,
+                                                property_iri=res['property_iri'],
+                                                property_value_iri=market_value_iri, 
+                                                property_value=market_value)
+        # Return graph with SPARQL update (empty for unavailable market value)
+        return g
+
+
+def default():
+    """
+        Instructional message at the app root.
+    """
+    # TODO: Update path to main upon merging
+    msg  = "This is an asynchronous agent to estimate the market value of a particular property (i.e. building, flat).<BR>"
+    msg += "<BR>"
+    msg += "For more information, please visit https://github.com/cambridge-cares/TheWorldAvatar/tree/dev-PropertyValueEstimationAgent/Agents/PropertyValueEstimationAgent<BR>"
```

Comment:
Please change this link to the `main` branch.

Reply:
Addressed by [70bf83d](https://github.com/cambridge-cares/TheWorldAvatar/pull/454/commits/70bf83da2a3c88d28f664d112439d09e18003cba)